src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,627 @@
+                    logger.warn("Probe successful, sent at "
+                            + bestProbe.getSendBitrateBps() +
+                            " bps, received at "
+                            + bestProbe.getRecvBitrateBps()
+                            + " bps. Mean send delta: " + bestProbe.sendMeanMs
+                            + " ms, mean recv delta: " + bestProbe.recvMeanMs
+                            + " ms, num probes: " + bestProbe.count);
+                    remoteRate.setEstimate(probeBitrateBps, nowMs);
+                    return probeResult.kBitrateUpdated;
+                }
+            }
+            // Not probing and received non-probe packet,
+            // or finished with current set  of probes.
+            if (clusters.size() >= kExpectedNumberOfProbes)
+                probes_.clear();
+            return probeResult.kNoUpdate;
+        }
+    }
+
+    boolean isBitrateImproving(int newBitrateBps)
+    {
+        synchronized (critSect)
+        {
+            boolean initialProbe = !remoteRate.isValidEstimate()
+                    && newBitrateBps > 0;
+            boolean bitrateAboveEstimate =
+                    remoteRate.isValidEstimate() &&
+                            newBitrateBps > (int) (remoteRate
+                                    .getLatestEstimate());
+            return initialProbe || bitrateAboveEstimate;
+        }
+    }
+
+
+    void incomingPacket( long arrival_time_ms, long payload_size,
+     RawPacket pkt)
+    {
+        //RTC_DCHECK(network_thread_.CalledOnValidThread());
+    /*if (!header.getExtension().hasAbsoluteSendTime) {
+        logger.warn( "RemoteBitrateEstimatorAbsSendTimeImpl: Incoming packet "
+        + "is missing absolute send time extension!";
+        return;
+    }
+    incomingPacketInfo(arrival_time_ms, header.getExtension().absoluteSendTime,
+            payload_size, header.ssrc);*/
+    }
+
+    void incomingPacketInfo(
+        long arrivalTimeMs,
+        long sendTime24bits,
+        long payloadSize,
+        long ssrc) {
+        //RTC_CHECK(sendTime24bits < (1L << 24));
+        if (!umaRecorded) {
+           // RTC_HISTOGRAM_ENUMERATION(kBweTypeHistogram,
+            // bweNames.kReceiverAbsSendTime, bweNames.kBweNamesMax);
+            umaRecorded = true;
+        }
+        // Shift up send time to use the full 32 bits that inter_arrival
+        // works with,
+        // so wrapping works properly.
+        long timestamp = sendTime24bits << kAbsSendTimeInterArrivalUpshift;
+        long sendTimeMs = (long) (timestamp * kTimestampToMs);
+        long nowMs = System.currentTimeMillis();
+        // TODO(holmer): SSRCs are only needed for REMB,
+        // should be broken out from  here.
+        // Check if incoming bitrate estimate is valid, and if it
+        // needs to be reset.
+        Long incomingBitrate_ =
+                incomingBitrate.getRate(arrivalTimeMs);
+        if (incomingBitrate_ != null)
```

Comment:
getRate returns a "long" (small "l"), and converting this to a "Long" (big "L") will always be non-null. Use a "long" and check for != 0 (or > 0 if appropriate).